src/Mapper/BaseGroupedMapper.php

```diff
@@ -131,6 +131,7 @@ public function with(string $name, array $options = []): self
                     'translation_domain' => $options['translation_domain'] ?? null,
                 ]); // add new tab automatically
             }
+            \assert(null !== $this->currentTab);
```

Comment:
Have you tried to make the with method final? Maybe it solves this. My main concern when adding asserts is that:

it is just to make phpstan happy

or

it is hiding a problem, because on some php configuration this will throw an error, on others it will just pass and fail somewhere else.

In our case I believe we are in the first case.

Reply:
We should indeed make the with method final. But still It's 100% to make phpstan happy.
Phpstan/psalm has no idea that the `with` call will set a currentTab.